### ISSUE TITLE
questions about softmax + cross_entropy v.s. softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by liurida** [aymericdamien/TensorFlow-Examples#50]
Hi, in the Logistic regression
the original code is 
```
pred=tf.nn.softmax(tf.add(tf.matmul(x,W),b))
cost =tf.reduce_mean(-tf.reduce_sum(y*tf.log(pred),reduction_indices=1))
```

we use cost to be minimized. 
But when I use the code below:
```
cross_entropy = tf.nn.softmax_cross_entropy_with_logits(tf.nn.softmax(tf.add(tf.matmul(x,W),b)),y)
cost=tf.reduce_mean(cross_entropy)
```
It could runs successfully, but the accuracy is not stable as the original one, it changed from 0.5 - 0.9.
I think the 2 code works as similar logic, but I could not get the essential cause for that.

**Comment by aymericdamien** [comment]
I think reason is that 'softmax_cross_entropy_with_logits' doesn't exactly apply cross entropy as defined in `cost =tf.reduce_mean(-tf.reduce_sum(y*tf.log(pred),reduction_indices=1))`. But this function is not so well documented, you can find back in the C++ sources how they compute it.

**Comment by liurida** [comment]
I find the error yesterday, maybe it's caused by cross_entropy = tf.nn.softmax_cross_entropy_with_logits(tf.nn.**softmax**(tf.add(tf.matmul(x,W),b)),y). I should not add another softmax in the softmax. I don't know the detail reason, but now the result is stable.

**Comment by aymericdamien** [comment]
Yes, that should be the mistake.
